### ISSUE TITLE
Bad performance trained on mnist/cifar10

### ISSUE THREAD

**Issue by StOnEGiggity** [StargazerZhu/YellowFin_MXNet#5]
I used main.py on mnist using one Titan X GPU.It seems that in the last epoch, the validation-accuracy is 0.8738 with loss is 0.44 with YellowFin and the validation-accuracy is 0.9677 with loss is 0.11.What confused me is that related to  Zehaos/MobileNet#27, it's fine in tensorflow.

I also trained resnet101 on cifar10 like article.I got bed performance, either.
Is there any idea for this question？Thanks a lot!

**Comment by rzhu3** [comment]
Thanks for your experiments and comments. Actually this is due to the different setting of rescale gradient scheme between MXNet and PyTorch, or maybe TensorFlow. In PyTorch, the loss function is the average loss of a batch, while in MXNet the loss function is usually defined as a finite-sum. Therefore, the "rescale_grad" should be set as "1.0/batch_size". In my experiments, the gradient variance does affect the performance, and I'm guessing normalization on input data may also help.

**Comment by StOnEGiggity** [comment]
Thanks again! It works on mnist. It may be one interesting question for better performance. I will try in my experience.
